### PR TITLE
fix:jupyter notebook rich format removal

### PR DESCRIPTION
**Pull request by SafetyQuincyF**
## Description

When safety auth is used with the `--headless`arg within a jyupter notebook, user's cannot see the url link to copy and added a disclaimer for line breaks splitting the url.

```
Copy and paste this URL into your browser:
⚠️ Ensure there are no extra spaces, especially at line breaks, as they may break the link.
https://example.com/cli/auth?response_type=code&client_id=XXXXXX&redirect_uri=https%3A%2F%2Fexample.com%2Fcli%2Fcallback&scope=openid+email+profile+offline_access&state=XXXXXX&sign_up=False&locale=en&ensure_auth=False&headless=True&code_challenge=XXXXXX&code_challenge_method=S256
Paste the response here:
```
## Type of Change

- [x] Bug fix
- [ ] New feature
- [ ] Documentation update
- [ ] Refactor
- [ ] Other (please describe):

## Testing

- [ ] Tests added or updated
- [x] No tests required

## Checklist

- [x] Code is well-documented
- [x] Changelog is updated (if needed)
- [x] No sensitive information (e.g., keys, credentials) is included in the code
- [ ] All PR feedback is addressed

## Summary by CodeRabbit

- **New Features**
	- Introduced a new debugging configuration for "Safety Auth Login --headless" to streamline the login process.
	- Added a function to check if the code is running in a Jupyter Notebook environment.

- **Bug Fixes**
	- Enhanced error handling and message formatting during the authentication process for clearer output.
	- Improved robustness in processing authentication callbacks and managing connection timeouts.

### CODE REVIEW

File: safety/auth/utils.py [hunk 1]
```diff
@@ -425,3 +426,20 @@ def send(self, request: requests.PreparedRequest, **kwargs: Any) -> requests.Res
...
+@lru_cache(maxsize=1)
+def is_jupyter_notebook() -> bool:
+    """
+    Check if the code is being executed inside a Jupyter Notebook.
+    
+    Returns:
+        bool: True if running in a Jupyter notebook, False otherwise.
+    """
+    try:
+         from IPython import get_ipython
```
Comment (dylanpulver) [inline safety/auth/utils.py:440]: Does it make sense to move this import to the top of the file?

File: safety/auth/utils.py [hunk 2]
```diff
@@ -425,3 +426,20 @@ def send(self, request: requests.PreparedRequest, **kwargs: Any) -> requests.Res
...
+         if 'IPKernelApp' not in get_ipython().config:
```
Comment (dylanpulver) [inline safety/auth/utils.py:441]: Here is an idea to slightly simplify things:
```
try:
        ipython_instance = get_ipython()
        return 'IPKernelApp' in ipython_instance.config
    except (ImportError, AttributeError):
        return False
```

**Review by dylanpulver — APPROVED** [review]
Looks good! Added 2 suggestions to look at for some minor potential improvements!

File: safety/auth/utils.py (hunk 1 above)
Comment (SafetyQuincyF) [reply safety/auth/utils.py:440]: Yes

Comment (SafetyQuincyF) [reply safety/auth/utils.py]: IPython will only be present in jyupter notebook env.

**Pull request merged**